src/Permeability/Permeability.jl

```diff
@@ -0,0 +1,198 @@
+    AbstractPermeability,
+    ConstantPermeability,        # constant
+    HazenPermeability,           # Hazen equation
+    PowerLawPermeability,        # Power-law permeability
+    CarmanKozenyPermeability    # Carman-Kozeny permeability
+
+# Define "empty" computational routines in case nothing is defined
+function compute_permeability!(
+    k::_T, s::No_MaterialParam{_T}; ϕ::_T=zero(_T)
+) where {_T}
+    return zero(_T)
+end
+function compute_permeability(s::No_MaterialParam{_T}; ϕ::_T=zero(_T)) where {_T}
+    return zero(_T)
+end
+
+# Constant Permeability
+"""
+    ConstantPermeability(k = 1e-12m^2)
+
+Defines a constant permeability value for a given material.
+
+# Arguments
+- `k::Float64`: The permeability value in square meters (m^2). Default is `1e-12 m^2`.
+
+# Example
+```julia
+rheology = SetMaterialParams(;
+                      Phase=1,
+                      CreepLaws=(PowerlawViscous(), LinearViscous(; η=1e21Pa * s)),
+                      Gravity=ConstantGravity(; g=9.81.0m / s^2),
+                      Density= MeltDependent_Density(),
+                      Permeability = ConstantPermeability(; k=1e-12m^2),
+                      )
+```
+"""
+@with_kw_noshow struct ConstantPermeability{_T,U} <: AbstractPermeability{_T}
+    k::GeoUnit{_T,U} = 1e-12m^2 # permeability
+end
+ConstantPermeability(args...) = ConstantPermeability(convert.(GeoUnit, args)...)
+isdimensional(s::ConstantPermeability) = isdimensional(s.k)
+
+@inline (s::ConstantPermeability)(; args...) = s.k.val
+@inline (s::ConstantPermeability)(args) = s(; args...)
+@inline compute_permeability(s::ConstantPermeability{_T}, args) where {_T} = s(; args...)
+@inline compute_permeability(s::ConstantPermeability{_T})       where {_T} = s()
+
+function param_info(s::ConstantPermeability)
+    return MaterialParamsInfo(; Equation=L"k = cst")
+end
+
+function compute_permeability!(k::AbstractArray, s::ConstantPermeability; kwargs...)
+    @unpack_val k_val = s
+    k[:] .= k_val
+    return nothing
+end
+
+function compute_permeability!(k::AbstractArray, s::ConstantPermeability, args)
+    return compute_permeability!(k, s; args...)
+end
+
+function show(io::IO, g::ConstantPermeability)
+    return print(io, "Constant permeability: k=$(UnitValue(g.k))")
+end
+
+# Hazen Permeability
+@with_kw_noshow struct HazenPermeability{_T,U1,U2} <: AbstractPermeability{_T}
+    C::GeoUnit{_T,U1} = 1.0 * NoUnits          # Hazen constant
+    D10::GeoUnit{_T,U2} = 1e-4 * m   # Effective grain size
+end
+HazenPermeability(args...) = HazenPermeability(convert.(GeoUnit, args)...)
+isdimensional(s::HazenPermeability) = isdimensional(s.D10)
+
+function param_info(s::HazenPermeability)
+    return MaterialParamsInfo(; Equation = L"k = C \cdot D_{10}^2")
+end
+
+function (s::HazenPermeability{_T})(; kwargs...) where {_T}
+    if kwargs isa Quantity
+        @unpack_units C, D10 = s
+    else
+        @unpack_val   C, D10 = s
+    end
+
+    return @pow C * D10^2
```

Comment:
```suggestion
    return C * D10^2
```

No need of using `@pow` for ^2